packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php

```diff
@@ -180,11 +180,11 @@ private function validateDuplication(
                 continue;
             }
 
-            if (in_array($duplicationPlaceholder, $this->reportedClassWithMethodDuplicate, true)) {
+            if (in_array($duplicationPlaceholder, self::$reportedClassWithMethodDuplicate, true)) {
                 continue;
             }
 
-            $this->reportedClassWithMethodDuplicate[] = $duplicationPlaceholder;
+            self::$reportedClassWithMethodDuplicate[] = $duplicationPlaceholder;
```

Comment:
`static` is not good idea as it breaks service design. This work around should be replace with clean solution.

Why is even reported twice? Other rules are reported just once.

It seems like this rule is registered twice.